src/main/scala/scorex/core/network/NodeViewSynchronizer.scala

```diff
@@ -188,11 +191,29 @@ class NodeViewSynchronizer[TX <: Transaction,
 
   //object ids coming from other node
   protected def processInv: Receive = {
-    case DataFromPeer(spec, invData: InvData@unchecked, remote)
+    case DataFromPeer(spec, invData: InvData@unchecked, peer)
       if spec.messageCode == InvSpec.MessageCode =>
 
-      //TODO can't replace viewHolderRef with a reader because of modifiers cache
-      viewHolderRef ! CompareViews(remote, invData._1, invData._2)
+      (mempoolReaderOpt, historyReaderOpt) match {
+        case (Some(mempool), Some(history)) =>
+          val modifierTypeId = invData._1
+          val modifierIds = modifierTypeId match {
+            case typeId: ModifierTypeId if typeId == Transaction.ModifierTypeId =>
+              mempool.notIn(invData._2)
+            case _ =>
+              invData._2.filter(mid => statusKeeper.status(mid)(history) == ModifiersStatusKeeper.Unknown)
```

Comment:
Once getting a wrongly formed modifier you do not have a chance to download it again.

Reply:
When we receive modifier with incorrect bytes, it is removed from status keeper (this line https://github.com/ScorexFoundation/Scorex/pull/280/files#diff-433d2867e69b9fe5dd369003650cac3fR241) since we do not know anything about real modifier with correct bytes.